src/LayerSwitcher/LayerSwitcher.tsx

```diff
@@ -0,0 +1,259 @@
+    if (!(_isEqual(this.props.layers, prevProps.layers))) {
+      this.setMapLayers();
+      this.updateLayerVisibilty();
+    }
+  }
+
+  /**
+   * Clones a layer
+   *
+   * @param layer The layer to clone.
+   * @returns The cloned layer.
+   */
+  cloneLayer = (layer: OlLayerTile | OlLayerGroup) => {
+    let layerClone: OlLayerTile | OlLayerGroup;
+    if (layer.getLayers) {
+      layerClone = new OlLayerGroup({
+        layers: layer.getLayers().getArray().map(this.cloneLayer),
+        originalLayer: layer,
+        ...layer.getProperties()
+      });
+    } else {
+      layerClone = new OlLayerTile({
+        source: layer.getSource(),
+        originalLayer: layer,
+        ...layer.getProperties()
+      });
+    }
+    return layerClone;
+  }
+
+  /**
+   * (Re-)adds the layers to the preview map and sets the visibleLayerIndex.
+   *
+   */
+  setMapLayers = () => {
+    const {
+      layers
+    } = this.props;
+    if (layers.length < 2) {
+      Logger.warn(`LayerSwitcher requires two or more layers.`);
+    }
+    this._map.getLayers().clear();
+    this._layerClones = layers.map((layer, index) => {
+      const layerClone = this.cloneLayer(layer);
+      if (layerClone.getVisible()) {
+        this._visibleLayerIndex = index;
+      }
+      this._map.addLayer(layerClone);
+      return layerClone;
+    });
+  }
+
+  /**
+   * Sets the visiblity of the layers in the props.map and this._map.
+   * Also sets the previewLayer in the state.
+   *
+   */
+  updateLayerVisibilty = () => {
+    const {
+      layers
+    } = this.props;
+    layers.forEach((l, i) => {
+      if (this._visibleLayerIndex === i) {
+        l.setVisible(true);
+      } else {
+        l.setVisible(false);
+      }
+    });
+    this._layerClones.forEach((l, i) => {
+      if (this._visibleLayerIndex === this._layerClones.length - 1 && i === 0) {
+        l.setVisible(true);
+        this.setState({ previewLayer: l });
+      } else if (this._visibleLayerIndex + 1 === i) {
+        l.setVisible(true);
+        this.setState({ previewLayer: l });
+      } else {
+        l.setVisible(false);
+      }
+    });
+  }
+
+  /**
+   * Constructs this._map
+   */
+  getMap = (): OlMap => {
+    const {
+      map
+    } = this.props;
+    return new OlMap({
+      view: map.getView(),
+      controls: []
+    });
+  }
+
+  /**
+   * Clickhandler for the overview switch.
+   *
+   */
+  onSwitcherClick = (evt: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+    evt.stopPropagation();
+    this._map.getLayers().getArray().forEach((layer, index) => {
+      if (layer.getVisible()) {
+        this._visibleLayerIndex = index;
+      }
+    });
+    this.updateLayerVisibilty();
+  }
+
+  /**
+   * The render function.
+   */
+  render() {
+    const {
+      className
+    } = this.props;
+
+    const finalClassName = className
+      ? `${className} ${this.className}`
+      : this.className;
+
+    return (
+      <div className={finalClassName}>
```

Comment:
Can we have some sort of a hint to users how to interact with the component? like a change of cursor, a tooltip like "Click to switch on the Sentinel-2 layer, 3 more layers available" or sth. like this?

Reply:
I don't think a tooltip is necessary. If you need one in your application you can still wrapp it around the component.